### ISSUE TITLE
How to implement a "Jump to Bottom" button?

### ISSUE THREAD

**Issue by rj-david** [chatscope/chat-ui-kit-react#70]
Is there an existing way to detect that the viewport of the message list is not at the bottom to be able to display a "Jump to Bottom" button?

![Screenshot from 2022-06-04 00-13-09](https://user-images.githubusercontent.com/30392688/171904794-5da9c3c0-4553-422f-90ea-3fc9fbb0d9c8.png)

**Comment by j-krl** [comment]
@KDederichs Yeah, good point. I thought there was access to the raw DOM methods but it looks like it's some kind of React wrapper.

What you could do is just use a getElementById to grab the messageList in your useEffect when a message comes in, and then you'll have access to all your regular DOM instance properties. It might be the child of the MessageList that you want -- the scrollbar-container I believe it is.

**Comment by KDederichs** [comment]
Kind of works yeah, though since it's an inner component you can't just slap an ID on it.... you actually have to do
```js
document?.getElementsByClassName(
    'scrollbar-container',
  )
```
and then take the fist one in there to get it... little bit annoying.

Well next thing to figure out would be how to properly keep the view position when adding elements to the top but tbh that's its own issue.

**Comment by TaiVuong** [comment]
Solution @supersnager , worked
- First, create a ref to hold message list
`const getListRef = () =>
    document.querySelector(
      ".cs-message-list__scroll-wrapper"
    );`

- Second, create a state to hold if scroll is bottom
`  const [isScrollAtBottom, setIsScrollAtBottom] = useState(true);
`

 - Next, using useEffect to hacky onScroll event for message list
useEffect(() => {
    const list = getListRef();
    if (list) {
      list.addEventListener('scroll', handleScroll);
    }
    return () => {
      if (list) {
        list.removeEventListener('scroll', handleScroll);
      }
    };
  },[ document.querySelector(
    ".cs-message-list__scroll-wrapper"
  )])

- Final, write a handleScroll function to logic check
`const handleScroll = (event:any) => {
    const list = getListRef();
    if(list) {
      if (Math.ceil(list.scrollTop) + list.clientHeight >= list.scrollHeight) {
        setIsScrollAtBottom(true);
      } else {
        setIsScrollAtBottom(false);
      }
    }
  };`
